utils/sockets.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __SOCKETS_H
+#define __SOCKETS_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Windows headers
+#ifdef __WIN32__
+#include <winsock2.h>
+#include <ws2tcpip.h>
+// ESP headers
+#elif __XTENSA__
```

Comment:
I haven't found one yet. I can't test for lwip defined macros at this point since lwip hasn't been included yet, so it's per platform really. At present, we have only discussed the Espressif SDK, which define `__XTENSA__`. Other platforms or SDKs will define their own.